From d1de409b0aeb8dffd42673444043caf01437804e Mon Sep 17 00:00:00 2001
From: Maxime Ripard <maxime.ripard@free-electrons.com>
Date: Mon, 13 Dec 2021 22:07:17 +0100
Subject: [PATCH] Modify the default lighttpd configuration file to have one a
 starting conf

  * Changed the log path to /var/log and logs filenames
  * Disable IPv6
  * Do not setuid to a user that doesn't exist on the system
  * Change the network backend to writev since linux-sendfile fails on buildroot

Signed-off-by: Maxime Ripard <maxime.ripard@free-electrons.com>
Signed-off-by: Simon Dawson <spdawson@gmail.com>
[Gustavo: update for 1.4.37]
Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>
---
 doc/config/conf.d/access_log.conf |  2 +-
 doc/config/lighttpd.conf          | 16 ++++++++--------
 2 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/doc/config/conf.d/access_log.conf b/doc/config/conf.d/access_log.conf
index e685d43..92cff4f 100644
--- a/doc/config/conf.d/access_log.conf
+++ b/doc/config/conf.d/access_log.conf
@@ -9,7 +9,7 @@ server.modules += ( "mod_accesslog" )
 ##
 ## Default access log.
 ##
-accesslog.filename          = log_root + "/access.log"
+accesslog.filename          = log_root + "/lighttpd-access.log"
 
 ##
 ## The default format produces CLF compatible output.
diff --git a/doc/config/lighttpd.conf b/doc/config/lighttpd.conf
index 495d2a1..c3e1446 100644
--- a/doc/config/lighttpd.conf
+++ b/doc/config/lighttpd.conf
@@ -13,8 +13,8 @@
 ## if you add a variable here. Add the corresponding variable in the
 ## chroot example as well.
 ##
-var.log_root    = "/var/log/lighttpd"
-var.server_root = "/srv/www"
+var.log_root    = "/var/log"
+var.server_root = "/var/www"
 var.state_dir   = "/run"
 var.home_dir    = "/var/lib/lighttpd"
 var.conf_dir    = "/etc/lighttpd"
@@ -90,7 +90,7 @@ server.port = 80
 ##
 ## Use IPv6?
 ##
-server.use-ipv6 = "enable"
+# server.use-ipv6 = "enable"
 
 ##
 ## bind to a specific IP
@@ -101,8 +101,8 @@ server.use-ipv6 = "enable"
 ## Run as a different username/groupname.
 ## This requires root permissions during startup. 
 ##
-server.username  = "lighttpd"
-server.groupname = "lighttpd"
+server.username  = "www-data"
+server.groupname = "www-data"
 
 ##
 ## Enable lighttpd to serve requests on sockets received from systemd
@@ -118,7 +118,7 @@ server.groupname = "lighttpd"
 ##
 ## Document root
 ##
-server.document-root = server_root + "/htdocs"
+server.document-root = server_root
 
 ##
 ## The value for the "Server:" response field.
@@ -144,7 +144,7 @@ server.pid-file = state_dir + "/lighttpd.pid"
 ##
 ## Path to the error log file
 ##
-server.errorlog             = log_root + "/error.log"
+server.errorlog             = log_root + "/lighttpd-error.log"
 
 ##
 ## If you want to log to syslog you have to unset the 
@@ -192,7 +192,7 @@ include conf_dir + "/conf.d/debug.conf"
 ## and write(). Every modern OS provides its own syscall to help network
 ## servers transfer files as fast as possible 
 ##
-#server.network-backend = "sendfile"
+server.network-backend = "writev"
 
 ##
 ## As lighttpd is a single-threaded server, its main resource limit is
-- 
2.34.1

